addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1581,6 +1582,442 @@ var FieldBinaryImage = AbstractFieldBinary.extend({
+     */
+    init: function (parent, signerName, options) {
+        options = (options || {});
+        options.title = _t("Adopt Your Signature");
+        options.size = 'medium';
+        options.technical = false;
+
+        options.buttons = [];
+        options.buttons.push({
+            text: _t("Adopt and Sign"), classes: "btn-primary", click: function (e) {
+                this.confirmFunction(this.signerName, this.$signatureField.jSignature("getData"));
+            }
+        });
+        options.buttons.push({text: _t("Cancel"), close: true, classes: "cancel"});
+
+        this._super(parent, options);
+        this.options = options || {};
+        this.signerName = signerName;
+        this.signatureRatio = 3.0;
+        this.signatureType = 'signature';
+        this.currentFont = 0;
+        this.mode = 'draw';
+    },
+    /**
+     * Create a font menu depends of his length. Hide the button to automatically sign if the signer name isn't know.
+     * If they are less than 2 font, the font selector is hidden.
+     * @override
+     */
+    start: function () {
+        this.$fontSelection = this.$(".o_sign_font_selection").first();
+
+        this.$modeButtons = this.$('a.o_sign_mode');
+        this.$signatureField = this.$(".o_sign_signature").first();
+        this.$fontDialog = this.$(".o_sign_font_dialog").first();
+        this.$fontSelection = this.$(".o_sign_font_selection").first();
+        for (var i = 0; i < SignatureDialog.fonts.length; i++) {
+            this.$fontSelection.append($("<a/>").data('fontNb', i).addClass('btn btn-block'));
+        }
+        this.$clearButton = this.$('.o_sign_clean').first();
+        this.$selectStyleButton = this.$('.o_sign_select_style').first();
+        this.$loadButton = this.$('.o_sign_load').first();
+
+        if (SignatureDialog.fonts.length < 2) {
+            this.$selectStyleButton.hide();
+        }
+        if (this.signerName === undefined) {
+            this.$modeButtons.filter("[data-mode='auto']").hide();
+        }
+
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Show a dialog
+     * @override
+     */
+    open: function () {
+        var self = this;
+        this.opened(function (e) {
+            var width = self.$signatureField.width();
+            var height = width / self.signatureRatio;
+
+            self.$signatureField.empty().jSignature({
+                'decor-color': 'transparent',
+                'background-color': '#FFF',
+                'color': '#000',
+                'lineWidth': 2,
+                'width': width,
+                'height': height
+            });
+
+            self.emptySignature = self.$signatureField.jSignature("getData");
+
+            self.$modeButtons.filter('.btn-primary').click();
```

Comment:
it is weird to me that you have to click a button of the dialog when you open it. You can probably set its state correctly without interacting with the DOM by calling the method the click would have called